### DISCUSSION TITLE
[Feature Request]: Custom Versioning Strategy

### DISCUSSION THREAD

**Discussion by AlexanderRichter** [microsoft/AL-Go#1596] [Ideas]
The VersioningStrategy allows do choose between different Versioning strategies.

It would be great to have the chance to also use a custom Versioning Strategy as another option.

My idea is to define a path and in the al-go settings to the custom powershell which would then basically override the logic from 
IncrementVersionNumber

**Comment by freddydk** [comment]
The VersioningStrategy is mostly determining the Build and Revision part of the build number, currently supporting 2 modes:

1: RUN_NUMBER.RUN_ATTEMPT
3: yyyyMMdd.HHmmss

In IncrementVersionNumber it is only used to determine whether you want to use the same version number for all projects.

Not sure there are enough different versioningstrategies to justify making it overridable - but maybe you can describe how you would like versioning to work?
